cpp/oneapi/dal/table/backend/homogen_table_impl.hpp

```diff
@@ -20,13 +20,84 @@
 
 namespace oneapi::dal::backend {
 
+enum class alloc_kind {
+    host, /// Non-USM pointer allocated on host
+    usm_host, /// USM pointer allocated by sycl::alloc_host
+    usm_device, /// USM pointer allocated by sycl::alloc_device
+    usm_shared /// USM pointer allocated by sycl::alloc_shared
+};
+
+#ifdef ONEDAL_DATA_PARALLEL
+inline alloc_kind alloc_kind_from_sycl(sycl::usm::alloc alloc) {
+    using error_msg = dal::detail::error_messages;
+    switch (alloc) {
+        case sycl::usm::alloc::host: return alloc_kind::usm_host;
+        case sycl::usm::alloc::device: return alloc_kind::usm_device;
+        case sycl::usm::alloc::shared: return alloc_kind::usm_shared;
+        default: throw invalid_argument{ error_msg::unsupported_usm_alloc() };
+    }
+}
+#endif
+
+#ifdef ONEDAL_DATA_PARALLEL
+inline sycl::usm::alloc alloc_kind_to_sycl(alloc_kind kind) {
+    switch (kind) {
+        case alloc_kind::usm_host: return sycl::usm::alloc::host;
+        case alloc_kind::usm_device: return sycl::usm::alloc::device;
+        case alloc_kind::usm_shared: return sycl::usm::alloc::shared;
+        case alloc_kind::host: return sycl::usm::alloc::unknown;
```

Comment:
Why do we need to have such a mapping from `host` to `unknown`? Why not simply remove this case and throw an exception in `default` case?